### DISCUSSION TITLE
Hard Delete

### DISCUSSION THREAD

**Discussion by chrisparkeronline** [FirelyTeam/firely-net-sdk#2063] [General]
Hello,

Using firely, how would one use the harddelete=true parameter?

**Comment by mbaltus** [comment]
If this is the parameter created by the Azure Api for FHIR to be able to delete the resource plus its complete history, you can use the FhirClient's Delete method like this:

```csharp
client.Delete("{resourcetype}/{id}?hardDelete=true");
```

Or, for a conditional delete, you can add to the SearchParams, and use that:
```csharp
var sp = new SearchParams().Add("identifier", "12345").Add("hardDelete", "true");
client.Delete("Patient", sp);
```

**Reply by chrisparkeronline** [reply]
Thank you very much!
